Zeebe client can't write credentials cache (when running in Docker)

__Describe the Bug__

When running the Connectors docker image built from any Dockerfile in this repo, the Zeebe client attempts to cache client credentials in a file located in `/home/connectors` and fails due to missing permissions.

This results in the following log messages:
```
io.grpc.StatusRuntimeException: CANCELLED
                <part of the stacktrace omitted>
Caused by: java.nio.file.AccessDeniedException: /home/camunda
        at java.base/sun.nio.fs.UnixException.translateToIOException(Unknown Source)
        at java.base/sun.nio.fs.UnixException.rethrowAsIOException(Unknown Source)
        at java.base/sun.nio.fs.UnixException.rethrowAsIOException(Unknown Source)
        at java.base/sun.nio.fs.UnixFileSystemProvider.createDirectory(Unknown Source)
        at java.base/java.nio.file.Files.createDirectory(Unknown Source)
        at java.base/java.nio.file.Files.createAndCheckIsDirectory(Unknown Source)
        at java.base/java.nio.file.Files.createDirectories(Unknown Source)
        at io.camunda.zeebe.client.impl.oauth.OAuthCredentialsCache.ensureCacheFileExists(OAuthCredentialsCache.java:147)
        at io.camunda.zeebe.client.impl.oauth.OAuthCredentialsCache.writeCache(OAuthCredentialsCache.java:74)
        at io.camunda.zeebe.client.impl.oauth.OAuthCredentialsCache.computeIfMissingOrInvalid(OAuthCredentialsCache.java:101)
        at io.camunda.zeebe.client.impl.oauth.OAuthCredentialsProvider.applyCredentials(OAuthCredentialsProvider.java:84)
        at io.camunda.zeebe.client.impl.ZeebeCallCredentials.lambda$applyRequestMetadata$0(ZeebeCallCredentials.java:48)
        ... 3 common frames omitted

```

The bug can be fixed by applying the following envioronment variable:
```ZEEBE_CLIENT_CONFIG_PATH=/tmp/connectors```

Note that `/tmp` should be used as home directories are write protected in the container file system.

__Steps to Reproduce__

1. Get or build the Connectors Docker image (e.g. build one locally from the Dockerfile in `bundle/default-bundle`)
2. Run a container against a self managed Camunda 8 installation (with keycloak)
3. Observe the messages above in the logs

__Expected Behavior__

Credentials are cached in a location that is not write protected, e.g. `/tmp/connectors`.


__Environment__

- Camunda version: 8.3.* / 8.4.0-alpha2

--- a/bundle/camunda-saas-bundle/Dockerfile
+++ b/bundle/camunda-saas-bundle/Dockerfile
@@ -11,5 +11,6 @@ COPY target/*-with-dependencies.jar /opt/app/
 RUN addgroup --gid 1001 camunda && useradd --no-create-home --gid 1001 --uid 1001 camunda
 USER 1001:1001
 
+ENV ZEEBE_CLIENT_CONFIG_PATH=/tmp/connectors
 # Using entry point to allow downstream images to add JVM arguments using CMD
 ENTRYPOINT ["java", "-cp", "/opt/app/*", "io.camunda.connector.runtime.saas.SaaSConnectorRuntimeApplication"]

--- a/bundle/default-bundle/Dockerfile
+++ b/bundle/default-bundle/Dockerfile
@@ -16,4 +16,5 @@ RUN chmod +x start.sh
 RUN addgroup --gid 1001 camunda && useradd --no-create-home --gid 1001 --uid 1001 camunda
 USER 1001:1001
 
+ENV ZEEBE_CLIENT_CONFIG_PATH=/tmp/connectors
 ENTRYPOINT ["/start.sh"]

--- a/connector-runtime/connector-runtime-application/Dockerfile
+++ b/connector-runtime/connector-runtime-application/Dockerfile
@@ -19,4 +19,5 @@ RUN addgroup --gid 1001 camunda && useradd --no-create-home --gid 1001 --uid 100
 USER 1001:1001
 
 # Use entry point to allow downstream images to add JVM arguments using CMD
+ENV ZEEBE_CLIENT_CONFIG_PATH=/tmp/connectors
 ENTRYPOINT ["/start.sh"]

